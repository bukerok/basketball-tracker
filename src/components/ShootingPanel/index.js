import { Fragment } from 'react';
import { Link } from 'react-router-dom';
import Typography from '@material-ui/core/Typography';
import Card from '@material-ui/core/Card';
import CardHeader from '@material-ui/core/CardHeader';
import CardContent from '@material-ui/core/CardContent';
import IconButton from '@material-ui/core/IconButton';
import ShareIcon from '@material-ui/icons/Share';
import ViewIcon from '@material-ui/icons/Visibility';

import { PAGES_ROOTS } from '../../helpers/navigation';

import './index.scss';

export default function ShootingPanel() {
  return (
    <Card className="shooting-panel">
      <CardHeader
        className="shooting-panel__header"
        title={<Typography variant="h6">Shooting</Typography>}
        action={
          <Fragment>
            <IconButton
              component={Link}
              to={PAGES_ROOTS.shooting}
            >
              <ViewIcon />
            </IconButton>
            <IconButton disabled>
              <ShareIcon />
            </IconButton>
          </Fragment>
        }
      />
      <CardContent>
        <div className="shooting-panel__info-container">
          <div className="info">
            <Typography
              variant="subtitle1"
              className="info__title"
            >FT</Typography>
            <Typography variant="body1">50.15%</Typography>
          </div>
          <div className="info">
            <Typography
              variant="subtitle1"
              className="info__title"
            >2PT</Typography>
            <Typography variant="body1">45%</Typography>
          </div>
          <div className="info">
            <Typography
              variant="subtitle1"
              className="info__title"
            >3PT</Typography>
            <Typography variant="body1">25.19%</Typography>
          </div>
        </div>
      </CardContent>
    </Card>
  );
};
