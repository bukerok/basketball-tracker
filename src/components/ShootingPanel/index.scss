.shooting-panel {
  display: flex;
  flex-direction: column;

  &__header {
    padding-bottom: 0;
  }

  &__info-container {
    display: grid;
    grid-template-columns: repeat(3, 1fr);

    .info__title {
      font-weight: 500;
    }
  }
}
