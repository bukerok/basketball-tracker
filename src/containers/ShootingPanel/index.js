import { useSelector } from 'react-redux';

import { selectRecords } from '../../store/features/shots/shotsSlice';
import { calculateStats, formatStat } from '../../helpers/statistics';
import StatsPanel from '../../components/StatsPanel';
import { PAGES_ROOTS } from '../../helpers/navigation';

import './index.scss';

export default function ShootingPanel() {
  const records = useSelector(selectRecords);
  const stats = calculateStats(records);

  return (
    <StatsPanel
      title="Shooting"
      viewUrl={PAGES_ROOTS.shooting}
      items={[
        {
          name: 'FT',
          value: formatStat(stats.ft),
        },
        {
          name: '2PT',
          value: formatStat(stats['2pt']),
        },
        {
          name: '3PT',
          value: formatStat(stats['3pt']),
        },
      ]}
    />
  );
};
