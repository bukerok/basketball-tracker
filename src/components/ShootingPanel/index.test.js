import renderer from 'react-test-renderer';

import ShootingPanel from '.';

jest.mock('react-router-dom', () => ({
  Link: 'a',
}));
jest.mock('react-redux', () => ({
  useSelector: () => [
    {
      date: 'mockDate',
      zone: 12,
      score: 3,
      attempts: 15,
    },
    {
      date: 'mockDate2',
      zone: 1,
      score: 5,
      attempts: 10,
    },
  ],
}));
jest.mock('../../store/features/shots/shotsSlice', () => ({
  selectRecords: () => {},
}));

describe('ShootingPanel', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<ShootingPanel />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
