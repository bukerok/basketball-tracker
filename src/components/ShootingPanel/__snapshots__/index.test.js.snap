// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShootingPanel renders correctly 1`] = `
<div
  className="MuiPaper-root MuiCard-root stats-panel MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardHeader-root stats-panel__header"
  >
    <div
      className="MuiCardHeader-content"
    >
      <h6
        className="MuiTypography-root MuiTypography-h6"
      >
        Shooting
      </h6>
    </div>
    <div
      className="MuiCardHeader-action"
    >
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiIconButton-root"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
        to="shooting"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
            />
          </svg>
        </span>
      </a>
      <button
        className="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
        disabled={true}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81 1.66 0 3-1.34 3-3s-1.34-3-3-3-3 1.34-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9c-1.66 0-3 1.34-3 3s1.34 3 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.16c-.05.21-.08.43-.08.65 0 1.61 1.31 2.92 2.92 2.92 1.61 0 2.92-1.31 2.92-2.92s-1.31-2.92-2.92-2.92z"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
  <div
    className="MuiCardContent-root stats-panel__info-container"
  >
    <div
      className="info"
    >
      <h6
        className="MuiTypography-root info__title MuiTypography-subtitle1"
      >
        FT
      </h6>
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        -
      </p>
    </div>
    <div
      className="info"
    >
      <h6
        className="MuiTypography-root info__title MuiTypography-subtitle1"
      >
        2PT
      </h6>
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        50.00%
      </p>
    </div>
    <div
      className="info"
    >
      <h6
        className="MuiTypography-root info__title MuiTypography-subtitle1"
      >
        3PT
      </h6>
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        20.00%
      </p>
    </div>
  </div>
</div>
`;
